src/main/java/com/rocketden/main/service/RoomService.java

```diff
@@ -298,4 +299,25 @@ public RoomDto updateRoomSettings(String roomId, UpdateSettingsRequest request)
         socketService.sendSocketUpdate(roomDto);
         return roomDto;
     }
+
+    public RoomDto setSpectator(String roomId, Boolean isSpectator, SetSpectatorRequest request) {
+        Room room = repository.findRoomByRoomId(roomId);
+
+        // Return error if room could not be found
+        if (room == null) {
+            throw new ApiException(RoomError.NOT_FOUND);
+        }
+
+        // Return error if the initiator is not the host or the same as the receiver
+        User initiator = UserMapper.toEntity(request.getInitiator());
+        User receiver = UserMapper.toEntity(request.getReceiver());
+        if (!room.getHost().equals(initiator) && !initiator.equals(receiver)) {
+            throw new ApiException(RoomError.INVALID_PERMISSIONS);
+        }
+
+        User modifiedUser = room.getUserByUserId(receiver.getUserId());
+        modifiedUser.setIsSpectator(isSpectator);
+
+        return RoomMapper.toDto(room);
```

Comment:
I think you need a call to `repository.save(room)` of sorts, otherwise this might not actually change the object in the database